py/selenium/webdriver/remote/webelement.py

```diff
@@ -313,25 +317,15 @@ def send_keys(self, *value):
             # file_input.send_keys(os.path.abspath("path/to/profilepic.gif"))
 
         """
+        ignore_files = kwargs.get('ignore_files', False)
         # transfer file to another machine only if remote driver is used
         # the same behaviour as for java binding
-        if self.parent._is_remote:
+        if self.parent._is_remote and not ignore_files:
             local_file = self.parent.file_detector.is_local_file(*value)
```

Comment:
Hmm, but ... doesn't look very pythonic to me. I have 2 points to this approach:

1) A side effect, affecting the behaviour of a method from different class (driver/element).
2) From programmer's point of view, if a method is named 'send_keys', it should send keys. File upload field handling should get its own method so the files can be handled accordingly. I know this cannot be changed so that is why I added the additional parameter that can make switching of the behaviour more practical and straightforward.